### PR TITLE
updates to gitlab ci flow after testing

### PR DESCRIPTION
**Pull request by ryan-cahill**
## Purpose
Updates to the GitLab CI example

## Changes
* Adds an extra warning for CI variables
* Pins `dind` image version
* Adds caching for image on deployments
* Allows failure for the destroy job
* Deletes the component and environment when the PR is approved

### CODE REVIEW

File: deployments/automated-previews.mdx [hunk 1]
```diff
@@ -151,59 +151,75 @@ jobs:
...
+deployed to the environment. When the PR is either merged or closed, the GitLab environment will be automatically deleted and the component deployed to the environment
+in the Architect Cloud will be destroyed.
+
+<Warning>
+  This example assumes that the repo has `ARCHITECT_EMAIL`, `ARCHITECT_PASSWORD`, `ARCHITECT_ACCOUNT`, and `ARCHITECT_CLUSTER` set as CI/CD variables.
+</Warning>
 
 <Warning>
   `ARCHITECT_PASSWORD` must be a [personal access
   token](https://cloud.architect.io/login).
```
Comment (Zach-at-Arch) [inline deployments/automated-previews.mdx:171]: Similar comment for the BitBucket - we want to ensure the user knows what the PAT is and how to generate one

File: deployments/automated-previews.mdx [hunk 2]
```diff
@@ -151,59 +151,75 @@ jobs:
...
+    - architect destroy --auto-approve
+    - architect environment:destroy "${ARCHITECT_ENVIRONMENT}" --auto-approve
   environment:
-    name: architect/preview-$CI_MERGE_REQUEST_ID
+    name: $ARCHITECT_ACCOUNT/preview-$CI_MERGE_REQUEST_IID
     action: stop
   rules:
     - if: $CI_MERGE_REQUEST_ID
+      allow_failure: true
       when: manual
```
Comment (Zach-at-Arch) [inline deployments/automated-previews.mdx:222]: Can't this be automated on PR close?

Comment (ryan-cahill) [reply deployments/automated-previews.mdx:222]: It is. This step just also gives the user a manual option, but we could remove it if you want

File: deployments/automated-previews.mdx (hunk 1 above)
Comment (ryan-cahill) [reply deployments/automated-previews.mdx:171]: https://gitlab.com/architect-io/product/-/issues/281

File: deployments/automated-previews.mdx (hunk 2 above)
Comment (Zach-at-Arch) [reply deployments/automated-previews.mdx:221]: Nope, all good - just wanted to ensure this is still automated and the user doesn't have to do anything

File: deployments/automated-previews.mdx (hunk 1 above)
Comment (ryan-cahill) [reply deployments/automated-previews.mdx]: Updated to include that link

**Pull request merged**